### PR TITLE
[CIR] Implement simple lowering of CIR to LLVMIR, currently only supp…

### PR DESCRIPTION
**Pull request by lanza**
…orting cir.ret

This patch adds the simple infrastructure necessary to enable the
`-emit-llvm` flag such that the following invocation will do what you'd
expect:

```
$ clang -cc1 -fenable-clangir -emit-llvm -o out.ll foo.c
```

There is only one operation currently supported, `cir.ret`, and it is
only supported for the case of returning nothing.

### CODE REVIEW

File: clang/lib/CIR/LowerToLLVM.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+std::unique_ptr<llvm::Module>
+lowerFromCIRToLLVMIR(mlir::ModuleOp theModule,
+                     std::unique_ptr<mlir::MLIRContext> mlirCtx,
+                     llvm::LLVMContext &llvmCtx) {
+  mlir::PassManager pm(mlirCtx.get());
+
+  pm.addPass(createLowerToLLVMIRPass());
+
+  // TODO: Handle this error
+  if (mlir::failed(pm.run(theModule)))
```
Comment (bcardosolopes) [inline clang/lib/CIR/LowerToLLVM.cpp:134]: Should this be an assert for now?

File: clang/lib/CIR/LowerToLLVM.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    ;
+
+  mlir::registerLLVMDialectTranslation(*mlirCtx);
+
+  llvm::LLVMContext llvmContext;
+  auto llvmModule = mlir::translateModuleToLLVMIR(theModule, llvmCtx);
+
+  // TODO: Handle this error
```
Comment (bcardosolopes) [inline clang/lib/CIR/LowerToLLVM.cpp:142]: Same here!

**Review by bcardosolopes — COMMENTED** [review]
Some post-commit review :)

File: clang/lib/CIR/LowerToLLVM.cpp (hunk 1 above)
Comment (lanza) [reply clang/lib/CIR/LowerToLLVM.cpp:134]: Done!

**Pull request merged**